components/brave_origin/brave_origin_policy_manager_unittest.cc

```diff
@@ -0,0 +1,288 @@
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_origin/brave_origin_policy_manager.h"
+
+#include "base/memory/raw_ptr.h"
+#include "base/strings/strcat.h"
+#include "base/test/task_environment.h"
+#include "brave/components/brave_origin/brave_origin_policy_info.h"
+#include "brave/components/brave_origin/brave_origin_utils.h"
+#include "brave/components/brave_origin/pref_names.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/testing_pref_service.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_origin {
+
+class BraveOriginPolicyManagerTest : public testing::Test {
+ public:
+  BraveOriginPolicyManagerTest() = default;
+  ~BraveOriginPolicyManagerTest() override = default;
+
+  void SetUp() override {
+    pref_service_ = std::make_unique<TestingPrefServiceSimple>();
+    pref_service_->registry()->RegisterDictionaryPref(kBraveOriginPolicies);
+  }
+
+  void TearDown() override {
+    // Reset the singleton for clean test state
+    if (manager_) {
+      manager_->initialized_ = false;
+      manager_->browser_policy_definitions_.clear();
+      manager_->profile_policy_definitions_.clear();
+      manager_->local_state_ = nullptr;
+      manager_->observers_.Clear();
+      manager_ = nullptr;
+    }
+  }
+
+ protected:
+  void CreateTestPolicy(BraveOriginPolicyMap& policies,
+                        const std::string& pref_name,
+                        bool default_value,
+                        BraveOriginPolicyScope scope) {
+    policies.emplace(
+        pref_name, BraveOriginPolicyInfo(
+                       pref_name, default_value, scope, true,
+                       base::StrCat({"Test", pref_name, "Policy"}), pref_name));
+  }
+
+  BraveOriginPolicyMap CreateBrowserTestPolicies() {
+    BraveOriginPolicyMap test_policies;
+    CreateTestPolicy(test_policies, "test.global.pref", false,
+                     BraveOriginPolicyScope::kGlobal);
+    return test_policies;
+  }
+
+  BraveOriginPolicyMap CreateProfileTestPolicies() {
+    BraveOriginPolicyMap test_policies;
+    CreateTestPolicy(test_policies, "test.profile.pref", true,
+                     BraveOriginPolicyScope::kProfile);
+    return test_policies;
+  }
+
+  void InitializeManager() {
+    manager_ = BraveOriginPolicyManager::GetInstance();
+    manager_->Init(CreateBrowserTestPolicies(), CreateProfileTestPolicies(),
+                   pref_service_.get());
+  }
+
+  void SetPolicyInLocalState(const std::string& key, bool value) {
+    base::Value::Dict policies_dict =
+        pref_service_->GetDict(kBraveOriginPolicies).Clone();
+    policies_dict.Set(key, value);
+    pref_service_->SetDict(kBraveOriginPolicies, std::move(policies_dict));
+  }
+
+  base::test::TaskEnvironment task_environment_;
+  std::unique_ptr<TestingPrefServiceSimple> pref_service_;
+  raw_ptr<BraveOriginPolicyManager> manager_ = nullptr;
```

Comment:
re: https://github.com/brave/brave-core/pull/31242#discussion_r2353882534

I was thinking this could just be a `std::unique_ptr<BraveOriginPolicyManager>` that you construct in the `SetUp` method. This would allow you to remove the manual resetting code in `TearDown`. This could make the unit test more robust in the event that we wish to add additional members to the policy manager, as the author will not need to update the reset code in this test.

Reply:
Actually, you might be able to get away with not using `std::unique_ptr` since the policy manager has a default constructor. The declaration could simply be: 

```
BraveOriginPolicyManager manager_;
```

Reply:
Since it's a singleton I think it's better for the tests to use the singleton